src/query/plan/read_write_type_checker.cpp

```diff
@@ -91,17 +91,24 @@ bool ReadWriteTypeChecker::PreVisit([[maybe_unused]] Foreach &op) {
 bool ReadWriteTypeChecker::Visit(Once &) { return false; }  // NOLINT(hicpp-named-parameter)
 
 void ReadWriteTypeChecker::UpdateType(RWType op_type) {
-  // Update type only if it's not the NONE type and the current operator's type
-  // is different than the one that's currently inferred.
-  if (type != RWType::NONE && type != op_type) {
-    type = RWType::RW;
-  }
   // Stop inference because RW is the most "dominant" type, i.e. it isn't
   // affected by the type of nodes in the plan appearing after the node for
   // which the type is set to RW.
   if (type == RWType::RW) {
     return;
   }
+
+  // if op_type is NONE, type doesn't change.
+  if (op_type == RWType::NONE) {
+    return;
+  }  
+
+  // Update type only if it's not the NONE type and the current operator's type
+  // is different than the one that's currently inferred.
+  if (type != RWType::NONE && type != op_type) {
+    type = RWType::RW;
+  }
+
```

Comment:
Nice fix indeed. Maybe consider adding a simple unit test. I used [this simple snippet](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxCAAzKQADqgKhE4MHt6%2BekkpjgJBIeEsUTHxdpgOaUIETMQEGT5%2BXLaY9nkM1bUEBWGR0XG2NXUNWc0KQ93BvcX9sQCUtqhexMjsHOaxwcjeWADUJrFuTuPEmKwH2CYaAIIbWzuY%2B4dOLIZ4CRdXtzeMPrvbTAUCl2ACUAOoAFQAnglHiYAOwWXahADyoWwpFBmLBmPB%2B3hABEDlYbl8Tl4HKDITDMG4EBUANbRFG0dD4knXXZc3YAN1QeDZcgS6CYBEw0NhEHBEseqASAH0CDS5uyvtz1bsAPSa3ZCkVi3ZK2G7AS0KG7PBUC0EMAcYEMVAEQ305Fo7CGmm7Qxsggu5DLU6CE2w4iiki2hRqjVc7Ue414YH4KhUaKMJ2%2BwzO2UhZ2iiP/ANps0Whgp4indAAOij0ctuwgRseYDABwJVJlIBAqPR%2BzMADZzH2402W7E23LFcrVTdo9HG0829KaZ3wcSaxqEUSZ7PY9U5SWy4xVrsoqIvEoqRbgb7Hixkk7zGZ0KgWMFDARH8PMXhK5hK9arwYW0P23aNYyYZMKjFNkInNG9hxNK0HSwYFgizXYEgMBgvQSWFamCYAvSoMViHQ5DMHXdVY34UiAHcEDwZAEHQ%2BcE12JR01QKlq1AjU6wbT1W1bdtlxAcEVQRDlZw1U4CGWBg1149VN0o7l%2BPnISxxE2FO27d1B0HYNJ2NZthKXHSuzdCTEVUudBK0idG0UzlZxU3i3JcrlzMeJzEW83S3RU2IOSCkKbnJSlvLpRlolCTBaOnTzeX5QVhVFcUaSlaljUcqdJNsrUdT3BID1TBhj1PJhz0ePE2Pgu9xl7J8XzfQRP0bb9f3/QhAOAgrwMghxMBguCXXnVAkIwTBUOw%2BDMMzJhcLOYgCKIkiyKm/qdRo3Z6MY5j4NY4EOMNLjwR4pL1XU%2ByiS0/yxLBaypOk7lZPk5zpI8raLStXL42BPTMXnZ9pqAjgnSYwxgD/Ar%2BL%2BuExzM7L2Es9EnoK9U3uIBTgoKr6lO5WM9XShDTXNOtCHzB10xdPSEO9dD/XLNNg2iMNiAjb6jt2JNDyDDNZpdAQfIQPM7QLZnBGLYIywrC7pOukzR0XZGAp7Az%2BwQ0yHIVRt0YJuzjSRjsHo%2B1zCS%2BWGrQEo3Ebu1XUf0/tDPh3ZtZVk29P1y7Zw0nXjIo3GDfxLcko89VvOHST7r00LLYtoPvmuMUWHmg0DjcRtmDYXYIWi5AmWIT4bkj/OGQ8Us8GALKZRLVIxFxZHdi8NKxW99U8/pAvon%2BLvC7NrkmJi4hKz9ttgnr2gB974fK2JsUZQgFv9UwOZp6x7Ch%2B7kenMT/HrnGUVGPYgh0E7e9TnOTW5TZogi8ODOIHGM%2BQAvs4WBdhRMTQBhGsjvXErqgULRQgTF6wAPysHUQF4Y5uhAIAl6XJkhPDcBnJqsczBmGntJCIl8GTYO5OHWc0CaoOxBPAyBPtozIIzmgz8XlHwEOjLgs4%2BDE6fQtlAwEpCTZggoTZYO6oaEP0OE1bkYJGHsMQSePBTCQ4FRIdpFG4J%2BHPUQcI1Boj6GrkwXI9ULCmBsLURuThYdTGY0wHJbGJpIx71MV8YITpXjBAgO3bkNEIBVSIHXRwYhJIaFCj4vAYgUG7EkMSXsVhLAT18bQNx0YdqeK8N45e6V/GBNSenUR4TgqROsJk1eCDEFeK4jE4JtB5S/BYAuE%2BR9kDymgR%2BQ43kLgQDKWINeUjinJK4gUypDA/jCUPo4epjSM4tNiNgJerdV56O5D/RqJSTSsnlKcBQXhaAPi0mXCuNBgDjORmXZkrJWntIqVUzEfSqmdOMdJBZTolkhFoqs6aGytlth2QIPZByZRHOIHFWipyGCT36T4S5MzQUsBuRjbkz9OxoGSSguhmCACSwLYkUMwUi0RZzIXYrcE1cFK9MVmHxc3CFVSyWSKRCyM%2BYjaGiM8OgF56zNlUt0bkgFJKyVPJZW89lWDclCFYOwelIiCUQCZXytlmk2y8rWW8iS4qT4v1cFPLpJjQ7KXsaYjgCxaCcAAKy8D8BwLQpBUCcFQZYaw7ElgrDhGYWIPBSAEE0HqhY9ImBYBiK40gDIQCGo0PoTgkgTXuotZwXgCgQDBrdWavVpA4CwBgIgFAL4Eh0GiOQSgaBU5ZpiACIwXBYgaGaFgHkjFMAADU8DxRRLCU1LqaCbOiDGiAEQI0RDfMQKEEa81sEECiBgZoI1YFeEYcQCbSD4FOJUHk00I2YFUBUZJaxzWONaBG2geBcG1ChB4LAEaCArRYJwHgCwqAGGAAoWt9bG3nt4PwQQIgxDsCkDIQQigVDqGnboZoWFjDWGsPoXdMbIALDlO0GNHAAC0z9WymBtZYMwGhdiwbBGIWg6GwSYAXb230q0MMOlg0hqJFgACcAAOUjGbRQ4ZI2R6w1HaOp1FLB2EIpBCMWja0KCaQXAMHcJ4Ro/ghM9CKCUbIyRUgCBGE0RIMn2gSb6DEMYfHKgCE6MMETowWhtCqBMFTMw1ODC6PJvQh86jGak1wBYCh7WrAkPqo14bp2Wo4LsVQVG%2Bywb7JIf4gHdglsrBoSsXB6y4EICQXszq5i8HjVoOYnqzg%2BsoAsANQaQ0cDDaQM9WXTXmo89G2Nrr3ULGTWmpYBAEjJJzRAPNmb6D/JFZwbzvn/OBahsF2IoXwu8GGtFlaZ9mjPuEKIcQH6xvfrUBG/9pBaKhgSI%2BlzHBjWkEK7wDzKJkm1adBNLzPm/MBaLYRELYWIsQA8Pm5rsW7MJfKyl71/Q/UGpy7wfLwbNuRo4CVuNj3/WBuDW92IbmitRrKwm5L2WzBg62xDxLHrSD4ZSM4SQQA%3D) to verify the functionality, something similar should be super easy to add.

Reply:
Hey @antaljanosbenjamin, do you maybe have an idea how to do this, since UpdateType is a private method, I don't want to make a friend class to test it. One thing I can do is use Operators like in other tests in `tests/unit/query_plan_read_write_typecheck.cpp`, but it seems messy to have to use different combinations of operators for all 16 cases.